src/language-js/print/array.js

```diff
@@ -171,6 +172,36 @@ function isConciselyPrintedArray(node, options) {
 }
 
 function printArrayElements(path, options, elementsProperty, print) {
+  const elements = path.node[elementsProperty];
+
+  function isLineAfterElementEmpty(elementNode) {
+    const text = options.originalText;
+    const elementIdx = elements.findIndex((v) => v == elementNode);
+    const nextNode = elements.find((e, i) => i > elementIdx && e !== null);
+
+    function skipIfComment(idx) {
+      const token = text.substring(idx, idx + 2);
+      if (token === "//") {
+        return idx + text.substring(idx).match(/\n/).index;
+      } else if (token === "/*") {
+        return idx + text.substring(idx).match(/\*\//).index;
+      } else {
+        return idx;
+      }
+    }
+    function skipParens(currentIdx, candidateIdx = currentIdx) {
+      if (!nextNode || currentIdx >= locStart(nextNode)) {
+        return candidateIdx;
+      } else {
+        const nextCandidate =
+          text[currentIdx] == ")" ? currentIdx + 1 : candidateIdx;
+        return skipParens(skipIfComment(currentIdx + 1), nextCandidate);
```

Comment:
Can `skip` function do the job?

Reply:
I think it can't replace `skipParens` (and I think I'll rename `skipParens` to `indexAfterFurthestParen` for clarity) however after looking into it `skipIfComment` can be implemented with `skip` calls which cuts down a fair bit of code, so I'll make that change